### PR TITLE
Add show field view functionality

### PR DESCRIPTION
**Pull request by b8ne**
As mentioned in [This Issue](https://github.com/Laravel-Backpack/CRUD/issues/226), this feature adds a user friendly 'show' view.  Its not 100% complete, but in a useable and non-breaking state so thought its a good time for a PR to get some further input.

Rather than basing it off columns, i have based it off fields - i think it is much easier on the eye, and there are far more field types than columns.

Functions have been placed in `PanelTraits\Show.php`.
With no developer input, `show_fields` will be filled from `setFromDb()`.

Basically all `field` views were cloned and updated to use here.  Because of this, tabs and attributes will still work.

All field add and remove methods were carried over, renamed with `show`. ie. `$this->crud->addShowFields()`.

If `setFromDb()` is not used, but show fields need to match edit fields, simply use `$this->crud->addShowFields($this->crud->getCurrentFields());` after `addField` declarations to avoid duped code.

Still @TODO: 

- update all `crud.show.fields.x` to be show fields and not inputs. (I have done a few such as text, ckeditor, select... the rest are just dupes from `fields`).

- add 'show' translation for all languages (only english is done at the moment).

### CODE REVIEW

File: src/PanelTraits/Show.php [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+
+        // if this is a relation type field and no corresponding model was specified, get it from the relation method
+        // defined in the main model
+        if (isset($complete_field_array['entity']) && ! isset($complete_field_array['model'])) {
+            $complete_field_array['model'] = $this->getRelationModel($complete_field_array['entity']);
+        }
+
+        // if the label is missing, we should set it
+        if (! isset($complete_field_array['label'])) {
+            $complete_field_array['label'] = ucfirst($complete_field_array['name']);
```
Comment (lloy0076) [inline src/PanelTraits/Show.php]: There's a running "issue" here where simply `ucfirst` may not be what people want (I've suggested a callback with a default in the past).

File: src/PanelTraits/Show.php [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+     *
+     * @param $field
+     * @return $this
+     */
+    public function addShowField($field)
+    {
+        // if the field_definition_array array is a string, it means the programmer was lazy and has only passed the name
+        // set some default values, so the field will still work
+        if (is_string($field)) {
+            $complete_field_array['name'] = $field;
```
Comment (lloy0076) [inline src/PanelTraits/Show.php]: That sounds more like `$field_definition` to me.

File: src/PanelTraits/Show.php [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+        // default fields are already handled in addField()
+        $this->show_fields[$complete_field_array['name']] = $complete_field_array;
+
+        return $this;
+    }
+
+    /**
+     * Add multiple fields to the show view
+     *
+     * @param $fields
```
Comment (lloy0076) [inline src/PanelTraits/Show.php]: Should that have a type?

File: src/PanelTraits/Show.php [hunk 4]
```diff
@@ -0,0 +1,175 @@
...
+     */
+    public function addShowFields($fields)
+    {
+        if (count($fields)) {
+            foreach ($fields as $field) {
+                $this->addShowField($field);
+            }
+        }
```
Comment (lloy0076) [inline src/PanelTraits/Show.php]: else ... what?

File: src/PanelTraits/Show.php [hunk 5]
```diff
@@ -0,0 +1,175 @@
...
+            ];
+        }
+
+        return $fields;
+    }
+
+    /**
+     * Allow access to the show crud functionality
+     */
+    public function enableShow() {
```
Comment (lloy0076) [inline src/PanelTraits/Show.php]: `{` on next line.

File: src/resources/views/show/fields/date_picker.blade.php [hunk 6]
```diff
@@ -0,0 +1,112 @@
...
...],yy:String(m).slice(2),yyyy:m,h:n%12||12,hh:d(n%12||12),H:n,HH:d(n),M:o,MM:d(o),s:p,ss:d(p),l:d(q,3),L:d(q>99?Math.round(q/10):q),t:n<12?"a":"p",tt:n<12?"am":"pm",T:n<12?"A":"P",TT:n<12?"AM":"PM",Z:g?"UTC":(String(e).match(b)||[""]).pop().replace(c,""),o:(r>0?"-":"+")+d(100*Math.floor(Math.abs(r)/60)+Math.abs(r)%60,4),S:["th","st","nd","rd"][j%10>3?0:(j%100-j%10!=10)*j%10]};return f.replace(a,function(a){return a in s?s[a]:a.slice(1,a.length-1)})}}();dateFormat.masks={default:"ddd mmm dd yyyy HH:MM:ss",shortDate:"m/d/yy",mediumDate:"mmm d, yyyy",longDate:"mmmm d, yyyy",fullDate:"dddd, mmmm d, yyyy",shortTime:"h:MM TT",mediumTime:"h:MM:ss TT",longTime:"h:MM:ss TT Z",isoDate:"yyyy-mm-dd",isoTime:"HH:MM:ss",isoDateTime:"yyyy-mm-dd'T'HH:MM:ss",isoUtcDateTime:"UTC:yyyy-mm-dd'T'HH:MM:ss'Z'"},dateFormat.i18n={dayNames:["Sun","Mon","Tue","Wed","Thu","Fri","Sat","Sunday","Monday","Tuesday","Wednesday","Thursday","Friday","Saturday"],monthNames:["Jan","Feb","Mar","Apr","May","Jun","Jul","Aug","Sep","Oct","Nov","Dec","January","February","March","April","May","June","July","August","September","October","November","December"]},Date.prototype.format=function(a,b){return dateFormat(this,a,b)};
```
Comment (lloy0076) [inline src/resources/views/show/fields/date_picker.blade.php]: Ok, what is that monstrosity? :(

**Comment by tabacitu** [comment]
@b8ne I'm sorry to tell you that I've decided I'm not going to merge this... I'm sorry for all the hard work you've put into it - it's obvious. But I don't really agree with the core concept - of using Fields instead of Columns.

I've recently merged @AbbyJanke 's PR - https://github.com/Laravel-Backpack/CRUD/pull/781 which uses Columns, because I think it's easier on the eye - and a way simpler implementation. I think creating new fields, that are actually readonly, would only make this package more difficult to maintain, and offer zero benefit for the user. Instead, he can just use the columns that he already knows, they exist and they do exactly the same thing. So it's more DRY, I think.

I'm going to close the PR, but please don't let this get your hopes down - you've contributed so much already and Backpack is better because of you :-) I encourage you to try the solution we've merged once it's public (later this week), I think you'll like it.

Cheers and thanks!

**Pull request closed without merging**